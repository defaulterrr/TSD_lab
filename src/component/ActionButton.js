import React from 'react'

class ActionButton extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            label:      props.label,
            action:     props.action
        }
        this.onClick = this.onClick.bind(this)
    }

    onClick = () => {
        this.state.action()
    }

    render() {

        return(
            <div className='ActionButton'>

        <button onClick={this.onClick}>{this.state.label}</button>
            </div>
        )

    }
}

export default ActionButton;