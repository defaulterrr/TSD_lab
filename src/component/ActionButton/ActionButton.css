.ActionButton {
    margin: 10px;
    display: inline;
}

.button {
    font-size: 12pt;
}