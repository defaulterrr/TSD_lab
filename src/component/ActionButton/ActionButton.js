import React from 'react'
import './ActionButton.css'

class ActionButton extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            label:      props.label,
            action:     props.action
        }
        this.onClick = this.onClick.bind(this)
    }

    onClick = () => {
        this.state.action(this.state.label)
    }

    render() {

        return(
            <div className='ActionButton' class="ActionButton">

        <button onClick={this.onClick} class="button">{this.state.label}</button>
            </div>
        )

    }
}

export default ActionButton;