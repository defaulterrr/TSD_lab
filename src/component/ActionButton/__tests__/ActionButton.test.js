import React from 'react'
import { create } from 'react-test-renderer'

import ActionButton from './../ActionButton'

describe("[RENDER] ActionButton â€“ Snapshot test", () => {
    test("Matches the snapshot", () => {
        const button = create(<ActionButton label="Action Button" action={()=>{}}/>)
        expect(button.toJSON()).toMatchSnapshot()
    })
})