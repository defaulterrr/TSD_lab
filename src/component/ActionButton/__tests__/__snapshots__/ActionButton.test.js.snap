// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] ActionButton â€“ Snapshot test Matches the snapshot 1`] = `
<div
  class="ActionButton"
  className="ActionButton"
>
  <button
    class="button"
    onClick={[Function]}
  >
    Action Button
  </button>
</div>
`;
